balancer/endpointsharding/endpointsharding.go

```diff
@@ -76,24 +75,28 @@ type endpointSharding struct {
 	mu sync.Mutex // Sync updateState callouts and childState recent state updates
 }
 
-// UpdateClientConnState creates a child for new endpoints and deletes children
-// for endpoints that are no longer present. It also updates all the children,
-// and sends a single synchronous update of the childrens' aggregated state at
-// the end of the UpdateClientConnState operation. If any endpoint has no
-// addresses, returns error without forwarding any updates. Otherwise returns
-// first error found from a child, but fully processes the new update.
-func (es *endpointSharding) UpdateClientConnState(state balancer.ClientConnState) error {
-	if len(state.ResolverState.Endpoints) == 0 {
+// ValidateEndpoints returns an error if the endpoints list is empty, or no
+// addresses are present in endpoint list.
```

Comment:
This comment should indicate how/why/when to use this, not what it does. 

```go
// The EndpointSharding balancer assumes ValidateEndpoints has been used to validate all Endpoints
// passed to UpdateCLientConnState.
```

?

Or is this scenario actually not a problem for the endpointsharding balancer itself?  In which case maybe this function belongs in package `resolver` or `balancer`?

Reply:
Yeah I can move it to resolver or balancer, as this should trigger failure case in all petiole policies.

Reply:
But also since it's gatekeeped in all *petiole* policies maybe it does make sense to keep it here? Although looks like we won't use this in petiole like ring hash so I'll move it to resolver.

Reply:
Updated comment as well, let me know what you think.